### ISSUE TITLE
Missing "Ask on Launch" executable value on LaunchAction for schemes

### ISSUE THREAD

**Issue by jmarq013** [tuist/tuist#6503]
### What problem or need do you have?

We're currently unable to define a scheme's LaunchAction (or ProfileAction) with the "Ask on Launch" value for the Executable property. On a given .xcscheme file this translates to `askForAppToLaunch = "Yes"` in the LaunchAction object. When an actual target reference is supplied, this setting is omitted from the .xcscheme file.

![Screenshot 2024-07-08 at 12 24 46 PM](https://github.com/tuist/tuist/assets/14981850/dee814fa-f43c-4f56-946b-6b4263f69ac6)

### Potential solution

Right now the [`executable`](https://docs.tuist.io/generated/manifest/structs/RunAction.html#executable) property takes in an optional `TargetReference` instance. We could potentially introduce a new parameter to the initializer for [`RunAction`](https://docs.tuist.io/generated/manifest/structs/RunAction.html#runaction) like `askForAppToLaunch` that is a boolean that overrides the `executable` property. There does seem to be a `None` option for this setting which is why passing a `nil` value for `executable` to have it equal `Ask on Launch` might not make sense.

### macOS version

14.5

### Tuist version

4.10.2

### Xcode version

15.4

**Comment by fortmarek** [comment]
Hey @jmarq013 👋 

You are right that this functionality is missing. If the `executable` is not specified, we default to the [first target from the `buildAction`](https://github.com/tuist/tuist/blob/main/Sources/TuistGenerator/Generator/SchemeDescriptorsGenerator.swift#L442).

I see two ways how to implement this:
- If `executable` is not specified, instead of defaulting to the first target in the `buildAction`, we'd default to `Ask on Launch`
- We extend `ProjectDescription` to explicitly configure `executable` to be `.askOnLaunch`

I am more leaning to the latter as the former could be a breaking behavior change for some projects out there. I'd consider doing the former in the next major Tuist version.

I think the API could be something similar to:
```swift
// ProjectDescription/Executable

public enum Executable {
  case askOnLaunch
  case executable(TargetReference?)
}
```

So you could do:
```swift
            runAction: .runAction(
                executable: .askOnLaunch
            )
```

To not make this breaking change, we'd need to introduce an extra static constructor of `RunAction`.

The [SchemeDescriptorsGenerator](https://github.com/tuist/tuist/blob/2d4fc701b8e3a185b104370c2ed6b0a803565ec0/Sources/TuistGenerator/Generator/SchemeDescriptorsGenerator.swift#L55) already support the `askOnLaunch` functionality, so the implementation would mostly consist of passing down the value from the `ProjectDescription` to the `SchemeDescriptorsGenerator`.

Would you want to try and implement the above? 🙂

**Comment by jmarq013** [comment]
Yes I'm interested in attempting to implement this 🙂 The enum approach definitely seems like the best method going forward but, like you called out, it might be a breaking change. I'll look into adding a new static constructor that introduces this new askOnLaunch parameter instead 👍

**Comment by FelixLisczyk** [comment]
Is there any update on this issue? I would also like to have this option. 👍

**Comment by fortmarek** [comment]
I'm not aware of any development around this – this issue is very much for grabs for anyone interested, my pointers around how to implement this still apply, too, so feel free to take this one! 😇

**Comment by FelixLisczyk** [comment]
I've opened a pair of PRs to implement this:

- **XcodeGraph**: https://github.com/tuist/XcodeGraph/pull/491 — adds `askForAppToLaunch` property to `RunAction` and `ProfileAction`
- **Tuist CLI**: https://github.com/tuist/tuist/pull/9373 — adds the `Executable` enum to `ProjectDescription` and wires it through to scheme generation

The API matches [the approach suggested above](https://github.com/tuist/tuist/issues/6503#issuecomment-2228088503):

```swift
runAction: .runAction(executable: .askOnLaunch)
profileAction: .profileAction(executable: .askOnLaunch)
```

Existing factory methods are unchanged for backward compatibility.
